sycl/source/device.cpp

```diff
@@ -37,13 +37,15 @@ device::device(const device_selector &deviceSelector) {
 
 vector_class<device> device::get_devices(info::device_type deviceType) {
   vector_class<device> devices;
+  // Host device availability should not depend on the forced type
+  const bool includeHost =
```

Comment:
Can we do something like " bool includeHost = deviceType | info::device_type::host" instead?

Reply:
We could do that by modifying the current underlying values of `cl::sycl::info::device_type` (and thus get rid of the `detail::match_types` helper function), but I think that's outside the scope of this patch.

Reply:
@sergey-semenov, could you open an issue for this refactoring, please?

Reply:
Will do